app/models/screening.rb

```diff
@@ -3,7 +3,18 @@
 class Screening < ApplicationRecord
   belongs_to :movie
   belongs_to :hall
+  has_many :reservations
   validates_associated :movie, :hall
   validates :start_time, :end_time, :price, presence: true
   validates :price, numericality: { greater_than: 0 }
+
+  def available_seats
+    taken_seats = []
+    reservations.each do |reservation|
+      reservation.tickets.each do |ticket|
+        taken_seats.push(ticket.seat)
+      end
```

Comment:
```suggestion
      all_taken_seats << reservation.tickets.pluck(:seat)
```

Reply:
this will return an array of arrays so then after the loop you can call taken_seats.flatten